### PR TITLE
Fix table name qualification

### PR DESCRIPTION
**Pull request by jgoizueta**
In some cases (when adding a new layer in builder, but not when creating a map from a dataset) the table names in MapConfig queries, for non-organization users, are not qualified with the schema (`public`). This works for most purposes since the `search_path` ini the database will include the public schema. But in some cases (the Geocoding analysis), the query ends up being passed as text to a database function, that for technical reasons (SECURITY DEFINER) needs to execute the query under a different search path.

This was partially handled in #15572 but we need to adjust table name qualification also in the front code.

### CODE REVIEW

**Comment by jgoizueta** [comment]
remove stag tag and bump assets version before merging!!

File: lib/assets/javascripts/builder/helpers/table-name-utils.js [hunk 1]
```diff
@@ -34,8 +34,11 @@ module.exports = {
     return name;
   },
 
-  getQualifiedTableName: function (tableName, userName, inOrganization) {
+  getQualifiedTableName: function (tableName, userName, inOrganization, force=false) {
```
Comment (jesusbotella) [inline lib/assets/javascripts/builder/helpers/table-name-utils.js:37]: I'd change `force` argument name to `forceSchema` or something like that

**Comment by jgoizueta** [comment]
Looking at the failing test, there's one thing that worries me:
dataset view urls paths like `dataset/xyz` now beccome `dataset/public.xyz`. I didn't expect to alter this, so I'll have to check how to avoid it.
But... if avoiding that is not easy: those paths with the schema actually work, so it's a cosmetic question whether we could allow for them.

File: lib/assets/javascripts/builder/helpers/table-name-utils.js (hunk 1 above)
Comment (jgoizueta) [reply lib/assets/javascripts/builder/helpers/table-name-utils.js]: :+1:

**Comment by jgoizueta** [comment]
I've detected some evil side effects of the inclusion of the schema in table names:

This one, mentioned before is not so evil, just ugly (but the link works)
Screenshot 2020-05-08 at 18 03 17

Oops: the layer is not recognized as a simple table layer, so it's not editable:
Screenshot 2020-05-08 at 18 05 47

Screenshot 2020-05-08 at 18 05 52

**Comment by jgoizueta** [comment]
As far as I can see this is working well now: I've changed the non-org users schema qualification of tables only for queries and checking for custom queries (which disables table editing) is working again.
But we should do some more thorough testing of the UI because there could be something I'm missing. (should check both with org and non-org users, custom queries and simple tables, with and without analysis, and check that tables are editable or not, the UI works well...)

**Comment by jgoizueta** [comment]
The do-not-merge badge is there to remember restoring the assets version and bumping it (I tend to forget).

**Comment by jgoizueta** [comment]
This is not ready yet, now when creating a map from a table it's not working correctly

**Comment by jesusbotella** [comment]
This is a difficult matter, as you say, because that string is used in a lot of places as the table identifier. I think we should just change it wherever we need it using the `force` parameter.

**Comment by jgoizueta** [comment]
Update:

With #15572 the initial layer of a map (created from a dataset for a non-org user) used public-schema-qualified named for tables in the analysis queries, which solved the geocoding problem (but not for other layers added in Builder). This change didn't alter que layer metadata options (table_name, query, sql_history), which is fortunate. But it altered the table name of the first layer in Builder which now included the `public` schema. This had the ill effect making the layer read-only (because Builder is somehow not getting the table permissions due to the name change).

With the changes in this PR the situation improves because:
* The table name for all layers remain as before (no public. prefix) and thus they are correctly recognised as editable.
* All layers have the default query used in analyses (and elsewhere) include the `public.` schema, so no geocoding problem for any layer.

But the change of query couldn't be done only for the queries inside analyses, but for default queries in general, because of two reasons:
* Otherwise the query is considered to be customised by user (differs from default) and the layer is again unnecessarily read-only.
* When entering the SQL edit (in Dashboard or Builder) the query should also be consistent to not be considered altered (so that the "Clear" button is displayed, and if cleared, the query with `public.` is saved in layer options as a customised query).

So this will have a visible effect for users: now the queries for non org users in the SQL editor appear with the `public.` prefix. And it also has an impact on in layer metadata options for new maps because the new `public.` queries will be saved in `query` and `sql_history`.

All in all I think the changes are not a problem and we should deploy this to fix the problems with geocoding and read-only layers. But let's keep an eye and be alert about possible problems that might arise when this is in production.

**Pull request merged**